### ISSUE TITLE
Add yarn

### ISSUE THREAD

**Issue by laggingreflex** [mhart/alpine-node#65]
Please add yarn!

https://medium.com/@mfornasa/using-yarn-with-docker-c116ad289d56#.a37dpjhmx

**Comment by mhart** [comment]
As much as I love yarn (it's really cool!) – the aim of this image is to be as small as possible, so I won't add it to the existing image. It's also unlikely that npm can be removed anytime soon, so I won't replace it with yarn either.

Best option would be to create your own image with yarn (and no npm perhaps) – or just install it as the first step in your Dockerfile

**Comment by ide** [comment]
I've opened an issue on the Yarn repo to publish it as an Alpine package: https://github.com/yarnpkg/yarn/issues/1326. If someone so inclined helped with that issue, it'd be pretty easy to add `RUN apk add --no-cache yarn` to your Dockerfile.

**Comment by tugis** [comment]
>it'd be pretty easy to add RUN apk add --no-cache yarn to your Dockerfile.

I tried this both with `mhart/alpine-node:6.9.1` and `mhart/alpine-node:base-6.9.1`. But I get:

```
Step 3 : RUN apk add --no-cache yarn
 ---> Running in fba68ebd12f3
fetch http://dl-cdn.alpinelinux.org/alpine/v3.4/main/x86_64/APKINDEX.tar.gz
fetch http://dl-cdn.alpinelinux.org/alpine/v3.4/community/x86_64/APKINDEX.tar.gz
ERROR: unsatisfiable constraints:
  yarn (missing):
    required by: world[yarn]
```

Anything missing from your solution @ide ?

**Comment by mhart** [comment]
@tugis it's not actually published as an Alpine package. @ide has just suggested it.

The easiest way to install is:

```
$ npm install --global yarn
```

**Comment by styfle** [comment]
Note that yarn was added to the official node images.

https://github.com/nodejs/docker-node/issues/243
https://github.com/nodejs/docker-node/pull/337

I'm not saying it should be added to this image, but just offering an alternative.

**Comment by mhart** [comment]
@styfle yeah I noticed that a while back – I thought it would add a lot more weight than that, but it's not too bad.

Will definitely consider for 8.x

**Comment by JigSawFr** [comment]
Love to see this in node alpine !

**Comment by wclr** [comment]
@mhart do you plan to create a separate image that includes yarn only without npm, or npm & yarn?

**Comment by mhart** [comment]
I'll just include yarn with npm in the main image for 8.x – now that Node.js has really grown in size, I think yarn will add only a little overhead, relatively.

**Comment by whitecolor** [comment]
Ok, this will be probably more suitable becase yarn does not fully reflects/implements npm capabilities (for example publishing).

**Comment by wclr** [comment]
Ok, this will be probably more suitable becase yarn does not fully reflects/implements npm capabilities (for example publishing).

When is it planed btw?

**Comment by mhart** [comment]
Sometime in the next month I'd say: https://github.com/nodejs/LTS#lts-schedule1

**Comment by wclr** [comment]
BTW consider settings by default `yarn config set no-progress true` to [disable progress bar](https://github.com/yarnpkg/yarn/pull/1962) unlikely needed in containers.

**Comment by mhart** [comment]
Included yarn in the latest v8.0.0 image – let me know if anyone has any problems! https://hub.docker.com/r/mhart/alpine-node/
